### ISSUE TITLE
Missing log_level for spanlogger.go output

### ISSUE THREAD

**Issue by lordslair** [grafana/loki#7443]
**Describe the bug**
Loki's own log lines doesn't always have a propre log_level set, especially output from `caller=spanlogger.go`

**To Reproduce**
1. Started Loki (SHA or version): `grafana/loki:2.4.1`
2. Started Promtail (SHA or version) : `grafana/promtail:2.4.1`
3. Query: `{} term`

**Expected behavior**
I expect `caller=spanlogger.go` logs to be displayed in logs with their log_level

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: YAML deployment

**Screenshots, Promtail config, or terminal output**
```
level=info ts=2022-10-17T16:05:20.839174749Z caller=checkpoint.go:575 msg="checkpoint done" time=4m30.29906301s
ts=2022-10-17T16:05:29.111078274Z caller=spanlogger.go:87 org_id=fake Ingester.TotalReached=1 Ingester.TotalChunksMatched=25 Ingester.TotalBatches=1 Ingester.TotalLinesSent=10 Ingester.HeadChunkBytes="1.8 MB" Ingester.HeadChunkLines=9311 Ingester.DecompressedBytes="42 MB" Ingester.DecompressedLines=225283 Ingester.CompressedBytes="5.1 MB" Ingester.TotalDuplicates=0 Store.TotalChunksRef=65 Store.TotalChunksDownloaded=65 Store.ChunksDownloadTime=5.474784ms Store.HeadChunkBytes="0 B" Store.HeadChunkLines=0 Store.DecompressedBytes="2.5 MB" Store.DecompressedLines=12022 Store.CompressedBytes="318 kB" Store.TotalDuplicates=0
ts=2022-10-17T16:05:29.111124566Z caller=spanlogger.go:87 org_id=fake Summary.BytesProcessedPerSecond="102 MB" Summary.LinesProcessedPerSecond=546483 Summary.TotalBytesProcessed="46 MB" Summary.TotalLinesProcessed=246616 Summary.ExecTime=451.278265ms
```

As you can see, the first line has proper log_level announced `info`
For all the lines following from `caller=spanlogger.go`, the level flag is not added to the log, making this field not automatically detected, and generating "unknown" log_level lines in the UI

I don't have the impression that it's due to some of the config I applied as it's quite basic and straight from the docs.
And I don't see this behaviour with any other callers in the Loki logs

**Comment by lordslair** [comment]
I upgraded to 2.6.1 both Loki and Promtail.
As far as I can see, it was fixed somewhere between 2.4.x and 2.6.x, so I close the issue

**Comment by tfabritz** [comment]
This happens to me on 2.7.0
